src/jestTransformer.ts

```diff
@@ -0,0 +1,64 @@
+import { createHash } from 'crypto';
+import { Transformer } from '@jest/transform';
+import { createTransformer, getCacheKey as getBabelCacheKey } from 'babel-jest';
+import graphQLTransformer from 'jest-transform-graphql';
+import { relative as pathRelative } from 'path';
+import { readFileSync } from 'fs';
+import loadConfig from './lib/load-config';
+import { createPaths } from './lib/paths';
+
+const isSchemaFile = (filePath: string) => filePath.endsWith('.graphqls');
+
+const jestTransformer: Transformer = {
+  getCacheKey(fileData, filename, configString, cacheKeyOptions) {
+    const babelCacheKey = getBabelCacheKey!(
+      fileData,
+      filename,
+      configString,
+      cacheKeyOptions as any,
+    );
+
+    return createHash('md5')
+      .update(babelCacheKey)
+      .update('graphql-let')
+      .digest('hex');
+  },
+  process(input, filePath, jestConfig, transformOptions) {
+    if (isSchemaFile(filePath)) {
+      return graphQLTransformer.process(
+        input,
+        filePath,
+        jestConfig,
+        transformOptions,
+      );
+    }
+
+    const { rootDir } = jestConfig;
+    const [config] = loadConfig(rootDir);
+
+    const { tsxFullPath } = createPaths(
+      rootDir,
+      pathRelative(rootDir, filePath),
+      config.cacheDir,
+    );
+
+    const tsxContent = readFileSync(tsxFullPath, 'utf-8');
+
+    const babelTransformer = createTransformer({
```

Comment:
Not sure but can we respect users' babel config? It's better if we can.

Reply:
Forgot to move this to a babel fixture.